workflowparsers/phonopy/calculator.py

```diff
@@ -93,15 +96,51 @@ def read_kpath(filename):
     return generate_kpath_parameters(points, [labels], npoints)
 
 
-def generate_kpath_ase(cell, symprec, logger=None):
+def test_non_canonical_hexagonal(cell: Cell, symprec: float) -> Optional[int]:
+    """
+    Tests if the cell is a non-canonical hexagonal cell
+    and returns the index of the ~ 60 degree angle (error range controlled by `symprec`).
+    """
     try:
```

Comment:
I will change this try by:
```
if not isinstance(cell, Cell):
    logger.warning("Input `cell` is not ase.cell.Cel object")
    return None
```

Reply:
From a performance perspective, the difference depends on how likely the exception is to trigger.
In low-probability cases `try-except` is better (this is also the Pythonic way). `if-else` performs better in higher probabilities for the undesirable outcome.

Reply:
Same as in the comment below.

Just do an initial if instance check and that's it.

Reply:
Both cases are quite different, so let's cover them separately.

I partially agree and disagree with you here.
When I design a function, I try to make it as standalone as possible. In this case, I'd argue that the failure mode of not having the right type passed is quite different from `None`, which here has the semantics: _no hexagonal cell was found and hence no rotation angle either_.
I agree with you that in this specific setup, both failure modes yield the same outcome, and the distinction is effectively void.

The "big" question is, when refactoring, will this semantic distinction become relevant again?

Reply:
We discussed in-person and agreed that `try` is fine, as long as we enforce proper typing.